src/DependencyInjection/DI/test/ServiceProviderContainerTests.cs

```diff
@@ -121,11 +121,17 @@ public void ResolvesServiceMixedServiceAndOptionalStructConstructorArguments()
         {
             var serviceCollection = new ServiceCollection();
             serviceCollection.AddSingleton<IFakeService, FakeService>();
-            serviceCollection.AddSingleton<ClassWithServiceAndOptionalArgsCtorWithStructs>();
+            serviceCollection.AddTransient<ClassWithServiceAndOptionalArgsCtorWithStructs>();
```

Comment:
Why not add a new test?

Reply:
Rationale was that I wasn’t testing a new behavior but a quirk in the existing implementation. In this context swapping to transient from singleton doesn’t affect what primary behavior we’re testing while enabling me to validate the quirk is fixed. 

I can easily fork the test to a separate one if you’d prefer.

Reply:
Yeah, fork it into a separate test and I wonder if we should verify all the parameter values.

Reply:
Done